From 32c5395c2a9abc2832fa7cbffcd92e3ea412fb47 Mon Sep 17 00:00:00 2001
From: Jan Engelhardt <jengelh@inai.de>
Date: Sat, 10 Nov 2018 15:54:48 +0100
Subject: [PATCH 58/58] treewide: fix google shitcode without returns

rpmlint exits non-zero:

[12111s] E: iridium-browser no-return-in-nonvoid-function ../../base/logging.h:381
[12111s] E: iridium-browser no-return-in-nonvoid-function ../../components/previews/content/previews_hints.cc:127
[12111s] E: iridium-browser no-return-in-nonvoid-function ../../components/signin/core/browser/about_signin_internals.cc:543
[12111s] E: iridium-browser no-return-in-nonvoid-function ../../components/signin/core/browser/profile_oauth2_token_service.cc:125
[12111s] E: iridium-browser no-return-in-nonvoid-function ../../media/blink/webmediaplayer_impl.cc:200
[12111s] E: iridium-browser no-return-in-nonvoid-function ../../media/cdm/cdm_helpers.cc:29
[12111s] E: iridium-browser no-return-in-nonvoid-function ../../net/third_party/quic/core/qpack/qpack_decoder.cc:172, 193, 285, 307
[12111s] E: iridium-browser no-return-in-nonvoid-function ../../third_party/blink/renderer/platform/wtf/casting.h:60
[12111s] E: iridium-browser no-return-in-nonvoid-function ../../third_party/webrtc/modules/video_coding/codecs/vp8/temporal_layers_checker.cc:30
[12111s] E: iridium-browser no-return-in-nonvoid-function ../../third_party/webrtc/modules/video_coding/codecs/vp8/vp8_temporal_layers.cc:40
[12111s] E: iridium-browser no-return-in-nonvoid-function ./../../chrome/browser/background_fetch/background_fetch_download_client.cc:40
[12111s] E: iridium-browser no-return-in-nonvoid-function ./../../chrome/browser/download/offline_item_model.cc:151
[12111s] E: iridium-browser no-return-in-nonvoid-function ./../../chrome/browser/download/offline_item_utils.cc:203
[12111s] E: iridium-browser no-return-in-nonvoid-function ./../../chrome/browser/signin/chrome_signin_client.cc:109
[12111s] E: iridium-browser no-return-in-nonvoid-function ./../../components/autofill/core/browser/autofill_external_delegate.cc:448
[12111s] E: iridium-browser no-return-in-nonvoid-function ./../../components/viz/service/display/skia_renderer.cc:1059
[12111s] E: iridium-browser no-return-in-nonvoid-function ./../../content/browser/media/audio_output_stream_broker.cc:57
[12111s] E: iridium-browser no-return-in-nonvoid-function ./../../content/renderer/media/stream/media_stream_constraints_util_video_device.cc:743, 755
[12111s] E: iridium-browser no-return-in-nonvoid-function ./../../extensions/browser/url_loader_factory_manager.cc:68
[12111s] E: iridium-browser no-return-in-nonvoid-function ./../../media/base/decode_status.cc:22
[12111s] E: iridium-browser no-return-in-nonvoid-function ./../../media/base/decryptor.cc:21
[12111s] E: iridium-browser no-return-in-nonvoid-function ./../../media/base/demuxer_stream.cc:21, 35
[12111s] E: iridium-browser no-return-in-nonvoid-function ./../../media/capture/video/video_capture_device.cc:87
[12111s] E: iridium-browser no-return-in-nonvoid-function ./../../media/filters/decoder_stream.cc:93
[12111s] E: iridium-browser no-return-in-nonvoid-function ./../../services/network/public/cpp/cors/cors.cc:541, 554, 571
[12111s] E: iridium-browser no-return-in-nonvoid-function ./../../services/network/public/cpp/host_resolver_mojom_traits.cc:61, 81
[12111s] E: iridium-browser no-return-in-nonvoid-function ./../../third_party/blink/renderer/core/layout/layout_flexible_box.cc:597, 612
[12111s] E: iridium-browser no-return-in-nonvoid-function ./../../third_party/blink/renderer/modules/accessibility/ax_layout_object.cc:1302
[12111s] E: iridium-browser no-return-in-nonvoid-function ./../../ui/gfx/skia_vector_animation.cc:166
---
 base/bind_internal.h                                         | 1 +
 .../background_fetch/background_fetch_download_client.cc     | 1 +
 chrome/browser/download/offline_item_model.cc                | 1 +
 chrome/browser/download/offline_item_utils.cc                | 1 +
 .../browser/media/webrtc/media_stream_capture_indicator.cc   | 1 +
 chrome/browser/resource_coordinator/tab_lifecycle_unit.cc    | 2 ++
 .../browser/ui/autofill/save_card_bubble_controller_impl.cc  | 1 +
 .../ui/views/autofill/local_card_migration_dialog_view.cc    | 5 +++++
 chrome/browser/ui/webui/discards/discards_ui.cc              | 1 +
 chrome/browser/ui/webui/media/media_engagement_ui.cc         | 1 +
 .../autofill/core/browser/autofill_external_delegate.cc      | 1 +
 components/autofill/core/browser/autofill_manager.cc         | 1 +
 components/previews/content/previews_hints.cc                | 1 +
 components/signin/core/browser/about_signin_internals.cc     | 1 +
 .../signin/core/browser/profile_oauth2_token_service.cc      | 1 +
 content/browser/media/audio_output_stream_broker.cc          | 1 +
 .../stream/media_stream_constraints_util_video_device.cc     | 2 ++
 extensions/browser/url_loader_factory_manager.cc             | 1 +
 gin/v8_isolate_memory_dump_provider.cc                       | 2 ++
 media/base/decode_status.cc                                  | 1 +
 media/blink/webmediaplayer_impl.cc                           | 1 +
 media/filters/decoder_stream.cc                              | 1 +
 services/network/public/cpp/cors/cors.cc                     | 3 +++
 services/network/public/cpp/host_resolver_mojom_traits.cc    | 4 ++++
 .../blink/renderer/core/layout/layout_flexible_box.cc        | 2 ++
 .../blink/renderer/modules/accessibility/ax_layout_object.cc | 1 +
 .../blink/renderer/platform/heap/heap_stats_collector.h      | 1 +
 third_party/blink/renderer/platform/wtf/casting.h            | 1 +
 ui/gfx/skia_vector_animation.cc                              | 1 +
 29 files changed, 42 insertions(+)

diff --git a/base/bind_internal.h b/base/bind_internal.h
index d8f574a4b335..ec0f46c19138 100644
--- a/base/bind_internal.h
+++ b/base/bind_internal.h
@@ -749,6 +749,7 @@ bool QueryCancellationTraitsImpl(BindStateBase::CancellationQueryMode mode,
           functor, std::get<indices>(bound_args)...);
   }
   NOTREACHED();
+  abort();
 }
 
 // Relays |base| to corresponding CallbackCancellationTraits<>::Run(). Returns
diff --git a/chrome/browser/background_fetch/background_fetch_download_client.cc b/chrome/browser/background_fetch/background_fetch_download_client.cc
index 69b75b6fdef7..12b62f9b530a 100644
--- a/chrome/browser/background_fetch/background_fetch_download_client.cc
+++ b/chrome/browser/background_fetch/background_fetch_download_client.cc
@@ -37,6 +37,7 @@ BackgroundFetchFailureReason ToBackgroundFetchFailureReason(
     case download::Client::FailureReason::CANCELLED:
       return BackgroundFetchFailureReason::CANCELLED;
   }
+  return BackgroundFetchFailureReason::CANCELLED;
 }
 
 }  // namespace
diff --git a/chrome/browser/download/offline_item_model.cc b/chrome/browser/download/offline_item_model.cc
index a7387fed4e3a..d84e73a42923 100644
--- a/chrome/browser/download/offline_item_model.cc
+++ b/chrome/browser/download/offline_item_model.cc
@@ -148,6 +148,7 @@ download::DownloadItem::DownloadState OfflineItemModel::GetState() const {
       NOTREACHED();
       return download::DownloadItem::CANCELLED;
   }
+  return download::DownloadItem::CANCELLED;
 }
 
 bool OfflineItemModel::IsPaused() const {
diff --git a/chrome/browser/download/offline_item_utils.cc b/chrome/browser/download/offline_item_utils.cc
index d9fce380427a..ada6af050d3a 100644
--- a/chrome/browser/download/offline_item_utils.cc
+++ b/chrome/browser/download/offline_item_utils.cc
@@ -173,6 +173,7 @@ OfflineItemUtils::ConvertFailStateToDownloadInterruptReason(
 #include "components/download/public/common/download_interrupt_reason_values.h"
 #undef INTERRUPT_REASON
   }
+  return FailState::NO_FAILURE;
 }
 
 // static
diff --git a/chrome/browser/media/webrtc/media_stream_capture_indicator.cc b/chrome/browser/media/webrtc/media_stream_capture_indicator.cc
index cc3c5f847223..bd30edd9b8e9 100644
--- a/chrome/browser/media/webrtc/media_stream_capture_indicator.cc
+++ b/chrome/browser/media/webrtc/media_stream_capture_indicator.cc
@@ -252,6 +252,7 @@ int& MediaStreamCaptureIndicator::WebContentsDeviceUsage::GetStreamCount(
       NOTREACHED();
       return video_stream_count_;
   }
+  return video_stream_count_;
 }
 
 MediaStreamCaptureIndicator::MediaStreamCaptureIndicator() {}
diff --git a/chrome/browser/resource_coordinator/tab_lifecycle_unit.cc b/chrome/browser/resource_coordinator/tab_lifecycle_unit.cc
index 0bc6cc017d4c..c76c2866b505 100644
--- a/chrome/browser/resource_coordinator/tab_lifecycle_unit.cc
+++ b/chrome/browser/resource_coordinator/tab_lifecycle_unit.cc
@@ -159,6 +159,7 @@ bool IsValidStateChange(LifecycleUnitState from,
       }
     }
   }
+  return false;
 }
 
 StateChangeReason DiscardReasonToStateChangeReason(
@@ -171,6 +172,7 @@ StateChangeReason DiscardReasonToStateChangeReason(
     case LifecycleUnitDiscardReason::URGENT:
       return StateChangeReason::SYSTEM_MEMORY_PRESSURE;
   }
+  return StateChangeReason::SYSTEM_MEMORY_PRESSURE;
 }
 
 struct FeatureUsageEntry {
diff --git a/chrome/browser/ui/autofill/save_card_bubble_controller_impl.cc b/chrome/browser/ui/autofill/save_card_bubble_controller_impl.cc
index f6e005a3b1dd..165484e58595 100644
--- a/chrome/browser/ui/autofill/save_card_bubble_controller_impl.cc
+++ b/chrome/browser/ui/autofill/save_card_bubble_controller_impl.cc
@@ -234,6 +234,7 @@ base::string16 SaveCardBubbleControllerImpl::GetWindowTitle() const {
       NOTREACHED();
       return base::string16();
   }
+  return {};
 }
 
 base::string16 SaveCardBubbleControllerImpl::GetExplanatoryMessage() const {
diff --git a/chrome/browser/ui/views/autofill/local_card_migration_dialog_view.cc b/chrome/browser/ui/views/autofill/local_card_migration_dialog_view.cc
index 6bfc3a5ab609..52cf606854f9 100644
--- a/chrome/browser/ui/views/autofill/local_card_migration_dialog_view.cc
+++ b/chrome/browser/ui/views/autofill/local_card_migration_dialog_view.cc
@@ -417,6 +417,7 @@ bool LocalCardMigrationDialogView::Accept() {
       controller_->OnDoneButtonClicked();
       return true;
   }
+  return false;
 }
 
 bool LocalCardMigrationDialogView::Cancel() {
@@ -429,6 +430,7 @@ bool LocalCardMigrationDialogView::Cancel() {
       controller_->OnViewCardsButtonClicked();
       return true;
   }
+  return false;
 }
 
 bool LocalCardMigrationDialogView::Close() {
@@ -492,6 +494,7 @@ void LocalCardMigrationDialogView::ConstructView() {
   } else {
     AddChildView(CreateFeedbackContentView(controller_, this).release());
   }
+  return {};
 }
 
 base::string16 LocalCardMigrationDialogView::GetOkButtonLabel() const {
@@ -504,6 +507,7 @@ base::string16 LocalCardMigrationDialogView::GetOkButtonLabel() const {
       return l10n_util::GetStringUTF16(
           IDS_AUTOFILL_LOCAL_CARD_MIGRATION_DIALOG_BUTTON_LABEL_DONE);
   }
+  return {};
 }
 
 base::string16 LocalCardMigrationDialogView::GetCancelButtonLabel() const {
@@ -516,6 +520,7 @@ base::string16 LocalCardMigrationDialogView::GetCancelButtonLabel() const {
       return l10n_util::GetStringUTF16(
           IDS_AUTOFILL_LOCAL_CARD_MIGRATION_DIALOG_BUTTON_LABEL_VIEW_CARDS);
   }
+  return {};
 }
 
 LocalCardMigrationDialog* CreateLocalCardMigrationDialogView(
diff --git a/chrome/browser/ui/webui/discards/discards_ui.cc b/chrome/browser/ui/webui/discards/discards_ui.cc
index 443409953613..694253c85d66 100644
--- a/chrome/browser/ui/webui/discards/discards_ui.cc
+++ b/chrome/browser/ui/webui/discards/discards_ui.cc
@@ -61,6 +61,7 @@ mojom::LifecycleUnitVisibility GetLifecycleUnitVisibility(
   NOTREACHED();
   return mojom::LifecycleUnitVisibility::VISIBLE;
 #endif
+  return mojom::LifecycleUnitVisibility::HIDDEN;
 }
 
 resource_coordinator::LifecycleUnit* GetLifecycleUnitById(int32_t id) {
diff --git a/chrome/browser/ui/webui/media/media_engagement_ui.cc b/chrome/browser/ui/webui/media/media_engagement_ui.cc
index ff5f4edc4aec..69a593e2e679 100644
--- a/chrome/browser/ui/webui/media/media_engagement_ui.cc
+++ b/chrome/browser/ui/webui/media/media_engagement_ui.cc
@@ -98,6 +98,7 @@ class MediaEngagementScoreDetailsProviderImpl
       case content::AutoplayPolicy::kDocumentUserActivationRequired:
         return "document-user-activation-required";
     }
+    return "";
   }
 
   const std::string GetPreloadVersion() {
diff --git a/components/autofill/core/browser/autofill_external_delegate.cc b/components/autofill/core/browser/autofill_external_delegate.cc
index 2f40e6d19118..e2c05893ad39 100644
--- a/components/autofill/core/browser/autofill_external_delegate.cc
+++ b/components/autofill/core/browser/autofill_external_delegate.cc
@@ -442,6 +442,7 @@ base::string16 AutofillExternalDelegate::GetSettingsSuggestionValue()
       NOTREACHED();
       return base::string16();
   }
+  return {};
 }
 
 }  // namespace autofill
diff --git a/components/autofill/core/browser/autofill_manager.cc b/components/autofill/core/browser/autofill_manager.cc
index 9abef350550b..73cbe0f19812 100644
--- a/components/autofill/core/browser/autofill_manager.cc
+++ b/components/autofill/core/browser/autofill_manager.cc
@@ -277,6 +277,7 @@ PopupType AutofillManager::GetPopupType(const FormData& form,
     default:
       NOTREACHED();
   }
+  return PopupType::kUnspecified;
 }
 
 bool AutofillManager::ShouldShowCreditCardSigninPromo(
diff --git a/components/previews/content/previews_hints.cc b/components/previews/content/previews_hints.cc
index c6b31fd24943..378809b24c16 100644
--- a/components/previews/content/previews_hints.cc
+++ b/components/previews/content/previews_hints.cc
@@ -123,6 +123,7 @@ base::Optional<PreviewsType> ConvertProtoOptimizationTypeToPreviewsType(
     case optimization_guide::proto::LITE_PAGE_REDIRECT:
       return PreviewsType::LITE_PAGE_REDIRECT;
   }
+  return base::nullopt;
 }
 
 net::EffectiveConnectionType ConvertProtoEffectiveConnectionType(
diff --git a/components/signin/core/browser/about_signin_internals.cc b/components/signin/core/browser/about_signin_internals.cc
index ad8dd0679f85..512434a10b5e 100644
--- a/components/signin/core/browser/about_signin_internals.cc
+++ b/components/signin/core/browser/about_signin_internals.cc
@@ -550,6 +550,7 @@ std::string AboutSigninInternals::RefreshTokenEvent::GetTypeAsString() const {
     case AboutSigninInternals::RefreshTokenEventType::kAllTokensLoaded:
       return "Loaded";
   }
+  return "";
 }
 
 AboutSigninInternals::SigninStatus::SigninStatus()
diff --git a/components/signin/core/browser/profile_oauth2_token_service.cc b/components/signin/core/browser/profile_oauth2_token_service.cc
index 50fb83dd5ce3..e0870200df17 100644
--- a/components/signin/core/browser/profile_oauth2_token_service.cc
+++ b/components/signin/core/browser/profile_oauth2_token_service.cc
@@ -220,6 +220,7 @@ bool ProfileOAuth2TokenService::HasLoadCredentialsFinishedWithNoErrors() {
       // Load credentials finished with success.
       return true;
   }
+  return false;
 }
 
 void ProfileOAuth2TokenService::RecreateDeviceIdIfNeeded() {
diff --git a/content/browser/media/audio_output_stream_broker.cc b/content/browser/media/audio_output_stream_broker.cc
index 2ee519d96de1..a26146489303 100644
--- a/content/browser/media/audio_output_stream_broker.cc
+++ b/content/browser/media/audio_output_stream_broker.cc
@@ -54,6 +54,7 @@ StreamBrokerDisconnectReason GetDisconnectReason(DisconnectReason reason,
     case DisconnectReason::kDefault:
       return StreamBrokerDisconnectReason::kDefault;
   }
+  return StreamBrokerDisconnectReason::kDefault;
 }
 
 }  // namespace
diff --git a/content/renderer/media/stream/media_stream_constraints_util_video_device.cc b/content/renderer/media/stream/media_stream_constraints_util_video_device.cc
index 7ee21280953c..e9f06633af11 100644
--- a/content/renderer/media/stream/media_stream_constraints_util_video_device.cc
+++ b/content/renderer/media/stream/media_stream_constraints_util_video_device.cc
@@ -555,6 +555,7 @@ blink::WebMediaStreamTrack::DisplayCaptureSurfaceType ToWebDisplaySurface(
     case media::mojom::DisplayCaptureSurfaceType::BROWSER:
       return blink::WebMediaStreamTrack::DisplayCaptureSurfaceType::kBrowser;
   }
+  return blink::WebMediaStreamTrack::DisplayCaptureSurfaceType::kWindow;
 }
 
 blink::WebMediaStreamTrack::CursorCaptureType ToWebCursorCaptureType(
@@ -567,6 +568,7 @@ blink::WebMediaStreamTrack::CursorCaptureType ToWebCursorCaptureType(
     case media::mojom::CursorCaptureType::MOTION:
       return blink::WebMediaStreamTrack::CursorCaptureType::kMotion;
   }
+  return blink::WebMediaStreamTrack::CursorCaptureType::kNever;
 }
 
 VideoDeviceCaptureCapabilities::VideoDeviceCaptureCapabilities() = default;
diff --git a/extensions/browser/url_loader_factory_manager.cc b/extensions/browser/url_loader_factory_manager.cc
index 5b8fc1fc6e17..3c660ef1ca7d 100644
--- a/extensions/browser/url_loader_factory_manager.cc
+++ b/extensions/browser/url_loader_factory_manager.cc
@@ -331,6 +331,7 @@ bool IsSpecialURLLoaderFactoryRequired(const Extension& extension,
     case FactoryUser::kExtensionProcess:
       return DoExtensionPermissionsCoverCorsOrCorbRelatedOrigins(extension);
   }
+  return false;
 }
 
 network::mojom::URLLoaderFactoryPtrInfo CreateURLLoaderFactory(
diff --git a/gin/v8_isolate_memory_dump_provider.cc b/gin/v8_isolate_memory_dump_provider.cc
index fe14263836c8..7843b8df45ef 100644
--- a/gin/v8_isolate_memory_dump_provider.cc
+++ b/gin/v8_isolate_memory_dump_provider.cc
@@ -133,6 +133,7 @@ std::string IsolateTypeString(IsolateHolder::IsolateType isolate_type) {
       return "utility";
   }
   LOG(FATAL) << "Unreachable code";
+  return "blargh";
 }
 
 bool CanHaveMultipleIsolates(IsolateHolder::IsolateType isolate_type) {
@@ -149,6 +150,7 @@ bool CanHaveMultipleIsolates(IsolateHolder::IsolateType isolate_type) {
       return false;
   }
   LOG(FATAL) << "Unreachable code";
+  return false;
 }
 
 }  // namespace anonymous
diff --git a/media/base/decode_status.cc b/media/base/decode_status.cc
index 110696fd9414..57a1872c905f 100644
--- a/media/base/decode_status.cc
+++ b/media/base/decode_status.cc
@@ -19,6 +19,7 @@ const char* GetDecodeStatusString(DecodeStatus status) {
     case DecodeStatus::DECODE_ERROR:
       return "DecodeStatus::DECODE_ERROR";
   }
+  return "BLARGH";
 }
 
 std::ostream& operator<<(std::ostream& os, const DecodeStatus& status) {
diff --git a/media/blink/webmediaplayer_impl.cc b/media/blink/webmediaplayer_impl.cc
index 442800b8a207..f352d49ad97d 100644
--- a/media/blink/webmediaplayer_impl.cc
+++ b/media/blink/webmediaplayer_impl.cc
@@ -200,6 +200,7 @@ EncryptionMode DetermineEncryptionMode(
     case EncryptionScheme::CIPHER_MODE_AES_CBC:
       return EncryptionMode::kCbcs;
   }
+  return EncryptionMode::kUnencrypted;
 }
 
 #if BUILDFLAG(ENABLE_FFMPEG)
diff --git a/media/filters/decoder_stream.cc b/media/filters/decoder_stream.cc
index 1659fcc6ae4e..c098ba684cb2 100644
--- a/media/filters/decoder_stream.cc
+++ b/media/filters/decoder_stream.cc
@@ -90,6 +90,7 @@ const char* GetStatusString(typename DecoderStream<StreamType>::Status status) {
     case DecoderStream<StreamType>::DECODE_ERROR:
       return "decode_error";
   }
+  return "blargh";
 }
 
 template <DemuxerStream::Type StreamType>
diff --git a/services/network/public/cpp/cors/cors.cc b/services/network/public/cpp/cors/cors.cc
index 88ebef77664e..fdc4eee98e83 100644
--- a/services/network/public/cpp/cors/cors.cc
+++ b/services/network/public/cpp/cors/cors.cc
@@ -540,6 +540,7 @@ bool IsCorsSameOriginResponseType(mojom::FetchResponseType type) {
     case mojom::FetchResponseType::kOpaqueRedirect:
       return false;
   }
+  return false;
 }
 
 bool IsCorsCrossOriginResponseType(mojom::FetchResponseType type) {
@@ -553,6 +554,7 @@ bool IsCorsCrossOriginResponseType(mojom::FetchResponseType type) {
     case mojom::FetchResponseType::kOpaqueRedirect:
       return true;
   }
+  return false;
 }
 
 bool CalculateCredentialsFlag(mojom::FetchCredentialsMode credentials_mode,
@@ -570,6 +572,7 @@ bool CalculateCredentialsFlag(mojom::FetchCredentialsMode credentials_mode,
     case network::mojom::FetchCredentialsMode::kInclude:
       return true;
   }
+  return false;
 }
 
 }  // namespace cors
diff --git a/services/network/public/cpp/host_resolver_mojom_traits.cc b/services/network/public/cpp/host_resolver_mojom_traits.cc
index 424cba438717..cf87f72e248c 100644
--- a/services/network/public/cpp/host_resolver_mojom_traits.cc
+++ b/services/network/public/cpp/host_resolver_mojom_traits.cc
@@ -233,6 +233,7 @@ DnsQueryType EnumTraits<DnsQueryType, net::DnsQueryType>::ToMojom(
     case net::DnsQueryType::SRV:
       return DnsQueryType::SRV;
   }
+  return ResolveHostParameters::DnsQueryType::UNSPECIFIED;
 }
 
 // static
@@ -259,6 +260,7 @@ bool EnumTraits<DnsQueryType, net::DnsQueryType>::FromMojom(
       *output = net::DnsQueryType::SRV;
       return true;
   }
+  return false;
 }
 
 // static
@@ -275,6 +277,7 @@ EnumTraits<ResolveHostParameters::Source, net::HostResolverSource>::ToMojom(
     case net::HostResolverSource::MULTICAST_DNS:
       return ResolveHostParameters::Source::MULTICAST_DNS;
   }
+  return ResolveHostParameters::Source::ANY;
 }
 
 // static
@@ -295,6 +298,7 @@ bool EnumTraits<ResolveHostParameters::Source, net::HostResolverSource>::
       *output = net::HostResolverSource::MULTICAST_DNS;
       return true;
   }
+  return false;
 }
 
 // static
diff --git a/third_party/blink/renderer/core/layout/layout_flexible_box.cc b/third_party/blink/renderer/core/layout/layout_flexible_box.cc
index 42fcaab49c56..59a99c0b65f0 100644
--- a/third_party/blink/renderer/core/layout/layout_flexible_box.cc
+++ b/third_party/blink/renderer/core/layout/layout_flexible_box.cc
@@ -641,6 +641,7 @@ LayoutUnit LayoutFlexibleBox::FlowAwareContentInsetBefore() const {
       return ContentInsetRight();
   }
   NOTREACHED();
+  abort();
 }
 
 DISABLE_CFI_PERF
@@ -656,6 +657,7 @@ LayoutUnit LayoutFlexibleBox::FlowAwareContentInsetAfter() const {
       return ContentLeft();
   }
   NOTREACHED();
+  abort();
 }
 
 LayoutUnit LayoutFlexibleBox::CrossAxisScrollbarExtent() const {
diff --git a/third_party/blink/renderer/modules/accessibility/ax_layout_object.cc b/third_party/blink/renderer/modules/accessibility/ax_layout_object.cc
index 8d0e93e1715d..edbd3dd1de59 100644
--- a/third_party/blink/renderer/modules/accessibility/ax_layout_object.cc
+++ b/third_party/blink/renderer/modules/accessibility/ax_layout_object.cc
@@ -1238,6 +1238,7 @@ ax::mojom::TextPosition AXLayoutObject::GetTextPosition() const {
     case EVerticalAlign::kSuper:
       return ax::mojom::TextPosition::kSuperscript;
   }
+  return AXNodeObject::GetTextPosition();
 }
 
 int AXLayoutObject::TextLength() const {
diff --git a/third_party/blink/renderer/platform/heap/heap_stats_collector.h b/third_party/blink/renderer/platform/heap/heap_stats_collector.h
index 2e8880d4ae43..50ba1eb4280c 100644
--- a/third_party/blink/renderer/platform/heap/heap_stats_collector.h
+++ b/third_party/blink/renderer/platform/heap/heap_stats_collector.h
@@ -95,6 +95,7 @@ class PLATFORM_EXPORT ThreadHeapStatsCollector {
       case Id::kVisitStackRoots:
         return "BlinkGC.VisitStackRoots";
     }
+    return "";
   }
 
   static constexpr int kNumScopeIds = kLastScopeId + 1;
diff --git a/third_party/blink/renderer/platform/wtf/casting.h b/third_party/blink/renderer/platform/wtf/casting.h
index ee2873f573b0..162b4f5a75ff 100644
--- a/third_party/blink/renderer/platform/wtf/casting.h
+++ b/third_party/blink/renderer/platform/wtf/casting.h
@@ -57,6 +57,7 @@ struct DowncastTraits {
   template <typename U>
   static bool AllowFrom(const U&) {
     static_assert(sizeof(U) == 0, "no downcast traits specialization for T");
+    return true;
   }
 };
 
diff --git a/ui/gfx/skia_vector_animation.cc b/ui/gfx/skia_vector_animation.cc
index b2e0f7f9bdf8..89986eef6254 100644
--- a/ui/gfx/skia_vector_animation.cc
+++ b/ui/gfx/skia_vector_animation.cc
@@ -164,6 +164,7 @@ float SkiaVectorAnimation::GetCurrentProgress() const {
       DCHECK(timer_control_);
       return timer_control_->GetNormalizedCurrentCycleProgress();
   }
+  return 0;
 }
 
 void SkiaVectorAnimation::Paint(gfx::Canvas* canvas,
-- 
2.21.0

